### ISSUE TITLE
Gossip to converge router schema caches faster

### ISSUE THREAD

**Issue by domodwyer** [influxdata/influxdb_iox#5005]
## TL;DR

We should do our best to ensure all the routers schema cache are kept up to date / in sync. This is because:

* The schema API is used by the query team to discover the schema of a table for querying
  * They need the correct schema, not a partial schema that hasn't converged yet
* Each schema cache miss results in a catalog query
  * This causes a catalog write amplification of N where N=number of routers that need to discover and cache the column
* We need a cheap way to enforce column table / column limits ("service protection limits")
  * See https://github.com/influxdata/influxdb_iox/issues/4062

Current behaviour is probably acceptable for the first two points. The third requirement is likely low priority for the time being.

## Technical Description of Current State

The routers cache the schemas of all tables. When a router starts up, it pre-warms this cache with all the schema elements in the catalog and then incrementally adds new schema elements to the cache as each router observes them in the writes it processes.

When a router sees a column that is not within its schema cache, it attempts to insert it to the catalog. This either succeeds if the column has never been observed by the overall system, or the insert fails if another router has previously observed the column. In either case, the router then adds this column to it's schema cache and uses it to avoid hitting the catalog for subsequent requests. 

This behaviour causes the routers to be eventually consistent, synchronising their state through the catalog: in the absence of new columns being wrote, and in the presence of writes driving the convergence, all the routers will eventually cache the full set of schema elements.

In practice, we see a low amount of entropy with the current system:

![image](https://user-images.githubusercontent.com/9275968/176688498-72128bed-1282-4532-84ee-30db1881f8a0.png)

## Proposal

Gossip schema additions between routers, providing a cheap, low latency way of converging their caches without hitting the catalog at all.

This will remove the catalog write amplification that drives convergence today.

**Comment by jacobmarble** [comment]
This idea would reduce catalog load, but we should hold off on catalog performance improvements while the catalog service is being developed.
